internal/tenants/manager_test.go

```diff
@@ -142,4 +147,17 @@ func TestTenantManagerGetTenant(t *testing.T) {
 	require.NotNil(t, tenant)
 	require.Equal(t, 0, tc.requestCount[t3.Id])
 	require.Equal(t, t3, *tenant)
+
+	// wait for tenants to expire
+	time.Sleep(cacheExpirationTime)
```

Comment:
Just a side thought: we could maybe use something like a `TestClock` to test cache expirations in general - instead of calling `time.Sleep` we would do something like `testClock.Advance(200 ms)` and it would validate that these expirations work without having to pause execution.

Then you don't have to consider reducing the timeout from `500ms` -> `200ms` in tests as well 😄

Reply:
You are absolutely right that sleeping in a test is not ideal, but mocking time is not so straightforward, as we would have to define an interface for it, have two different implementations, one for std and one for our "test time", and have an instance for it in the cache, so it gets kind of cumbersome. For other cases where it's required, such as when using a ticker, I usually do [this](https://github.com/grafana/synthetic-monitoring-agent/blob/main/internal/telemetry/ticker.go), but for this case sleeping some time seems reasonable :confused: